test/unit/local/steps/initial_diff.js

```diff
@@ -0,0 +1,100 @@
+  let builders
+  let buffer
+
+  before('instanciate config', configHelpers.createConfig)
+  beforeEach('instanciate pouch', pouchHelpers.createDatabase)
+  beforeEach('populate pouch with documents', function () {
+    builders = new Builders({pouch: this.pouch})
+    buffer = new Buffer()
+  })
+  afterEach('clean pouch', pouchHelpers.cleanDatabase)
+  after('clean config directory', configHelpers.cleanConfig)
+
+  it('detects documents moved while client was stopped', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('baz').ino(2).create()
+
+    const bar = builders.event().action('scan').kind('directory').path('bar').stats({ ino: 1 }).build()
+    const buz = builders.event().action('scan').kind('file').path('buz').stats({ ino: 2 }).build()
+    buffer.push([bar, buz])
+    buffer = initialDiff(buffer, { pouch: this.pouch })
+
+    const events = await buffer.pop()
+    should(events).deepEqual([
+      builders.event(bar).action('renamed').oldPath('foo').build(),
+      builders.event(buz).action('renamed').oldPath('baz').build()
+    ])
+  })
+
+  it('detects documents moved while client is doing initial scan', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('foo/baz').ino(2).create()
+    builders.metadir().path('bar').ino(3).create()
+
+    const foo = builders.event().action('scan').kind('directory').path('foo').stats({ ino: 1 }).build()
+    const barbaz = builders.event().action('created').kind('file').path('bar/baz').stats({ ino: 2 }).build()
+    buffer.push([foo, barbaz])
+    const bar = builders.event().action('scan').kind('directory').path('bar').stats({ ino: 3 }).build()
+    buffer.push([
+      bar,
+      builders.event().action('scan').kind('file').path('bar/baz').stats({ ino: 2 }).build()
+    ])
+    buffer = initialDiff(buffer, { pouch: this.pouch })
+
+    const events = [].concat(
+      await buffer.pop(),
+      await buffer.pop()
+    )
+    should(events).deepEqual([
+      foo,
+      builders.event(barbaz).action('renamed').oldPath('foo/baz').build(),
+      bar
+    ])
+  })
+
+  it('detects documents replaced by another one of a different kind while client was stopped', async function () {
+    builders.metadir().path('foo').ino(1).create()
+    builders.metafile().path('bar').ino(2).create()
+    builders.metadir().path('baz').ino(3).create()
+    builders.metafile().path('buz').ino(4).create()
+
+    const foo = builders.event().action('scan').kind('file').path('foo').stats({ ino: 2 }).build()
+    const buz = builders.event().action('scan').kind('directory').path('buz').stats({ ino: 3 }).build()
+    buffer.push([foo, buz])
+    buffer = initialDiff(buffer, { pouch: this.pouch })
+
+    const events = await buffer.pop()
+    should(events).deepEqual([
+      builders.event(foo).action('renamed').oldPath('bar').build(),
+      builders.event(buz).action('renamed').oldPath('baz').build()
```

Comment:
Shouldn't we have `deleted` events for `foo` and `buz` preceding the `renamed` ones?

Reply:
No we only have the `renamed` event with the new watcher.

Reply:
Won't this generate conflits for both [files](https://github.com/cozy-labs/cozy-desktop/blob/master/core/merge.js#L281-L282) and [folders](https://github.com/cozy-labs/cozy-desktop/blob/master/core/merge.js#L319-L323)?

Reply:
Possibly. I didn't think much about the implications of the code I was testing on other parts of the application.
Maybe we just need to mark those events as overwriting?

Reply:
Actually, I believe it won't be an issue for [files](https://github.com/cozy-labs/cozy-desktop/blob/9a1d05f50dace7e953b5e74ed2c70fea428bc2eb/core/merge.js#L278-L280) but will for folders as we don't have the equivalent branch in `moveFolderAsync` :/.

Reply:
A separate issue was extracted, it will be fixed in a future PR.